lib/arel/nodes/casted.rb

```diff
@@ -11,7 +11,7 @@ def initialize val, attribute
       def nil?; @val.nil?; end
 
       def hash
-         [@class, @val, @attribute].hash
+         [@val, @attribute].hash
```

Comment:
We should change it to `self.class` like binary https://github.com/vipulnsward/arel/blob/171b422386f0e7183993ab554b152b7f310eb8df/lib/arel/nodes/binary.rb